### DISCUSSION TITLE
How to verify credentials in CredentialDialog ?

### DISCUSSION THREAD

**Discussion by mridah** [ookii-dialogs/ookii-dialogs-wpf#4] [Q&A] [answered]
I want to verify the credentials entered against a windows user credentials ( basically check if the user knows any windows user's username and password )

How can I accomplish this ?

    try
                {
                    // The window title will not be used on Vista and later; there the title will always be "Windows Security".
                    dialog.WindowTitle = "Enter windows credentials";
                    dialog.MainInstruction = "Please enter your username and password.";
                    dialog.Content = "In order to reset your password, you need to enter your Windows user name and password.";
                    dialog.ShowSaveCheckBox = false;
                    dialog.ShowUIForSavedCredentials = true;
                    // The target is the key under which the credentials will be stored.
                    // It is recommended to set the target to something following the "Company_Application_Server" pattern.
                    // Targets are per user, not per application, so using such a pattern will ensure uniqueness.
                    dialog.Target = "Ookii_DialogsWpfSample_www.example.com";
                    if (dialog.ShowDialog(this))
                    {                        
                        dialog.ConfirmCredentials(true);
                        MessageBox.Show("verified");
                    }
                }
                catch(Exception err) when (
                err is CredentialException
                || err is InvalidOperationException)
                {
                    MessageBox.Show("not verified");
                }

This always shows "not verified"

**Comment by augustoproiete** [comment, marked as answer]
@mridah The `CredentialDialog` class is meant to be a generic user/password **dialog** that you can use to capture the user's login and password and then _you_ validate it against any authentication system you want...

You will have to write the code for validating your credentials against your Active Directory, or some other authentication system you have...

For example, if you want to validate against a Active Directory in your domain, you 

```csharp
using System.DirectoryServices.AccountManagement;

void Main()
{
	using (PrincipalContext pc = new PrincipalContext(ContextType.Domain, "YOUR-DOMAIN-NAME"))
	{
		bool isValid = pc.ValidateCredentials("user-login", "user-password");
		
		// ...
	}
}

```

You can find more examples on [StackOverflow](https://stackoverflow.com/questions/290548/validate-a-username-and-password-against-active-directory/499716).
